app/models/topology.rb

```diff
@@ -92,4 +93,14 @@ def validate_stakeholders
       end
     end
   end
+
+  def validate_clone
+    return unless original.present?
+    return unless each_node
+
+    if each_node.count != original.each_node.count
```

Comment:
[Prefer `#length` or `#size` over `#count`](https://github.com/bbatsov/ruby-style-guide#count-vs-size). To be fair, it makes no different in this situation as `each_node` without args returns an `Enumerator`, but it's a good habit to be in.